Source/Core/ManagedArray.js

```diff
@@ -0,0 +1,149 @@
+/*global define*/
+define([
+        './defaultValue',
+        './defined',
+        './defineProperties',
+        './Check'
+    ], function(
+        defaultValue,
+        defined,
+        defineProperties,
+        Check) {
+    'use strict';
+
+    /**
+     * A wrapper around arrays so that the internal length of the array can be manually managed.
+     *
+     * @alias ManagedArray
+     * @constructor
+     * @private
+     *
+     * @param {Number} [length=0] The initial length of the array.
+     */
+    function ManagedArray(length) {
```

Comment:
I don't plan on reviewing this PR, but I also don't see the point of this class.  How is this any different than a raw array with a tracking index or the Queue that we already have?  Not trying to bikeshed, but every new class we add the Cesium needs to be maintained.